src/host.rs

```diff
@@ -192,6 +192,15 @@ impl<'a> HostAndPort<&'a str> {
     }
 }
 
+impl<S: AsRef<str>> fmt::Display for HostAndPort<S> {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        try!(self.host.fmt(f));
```

Comment:
Yes, sorry, on rereading that was confusing.

I'm suggesting that if possible we do a single, atomic, write to the formatter. Right now we might write `host:` and then something bad happens and writing the port fails so we end up with a half written thing.

It's probably not actually an issue.